### PR TITLE
fix(web): Inject latest Clerk JWT via Axios interceptor

### PR DESCRIPTION
**Pull request by SokratisVidros**
### What changed? Why was the change needed?

Fetching the Clerk JWT should always be done via the Clerk SDK, not the __session cookie that acts as a cache and creates the risk of getting an outdated JWT.

### Screenshots
image (1)

### CODE REVIEW

File: apps/web/src/api/api.client.ts [hunk 1]
```diff
@@ -78,46 +59,42 @@ function buildUrl(url: string, absoluteUrl: boolean) {
...
-        'Novu-Environment-Id': lastEnvironmentId || '',
-      }
-    : {};
-}
-
 // WIP: The static API client needs to be replaced by a dynamic API client where api keys are injected.
 export function buildApiHttpClient({
   baseURL = API_ROOT || 'https://api.novu.co',
   secretKey,
   jwt,
```
Comment (ChmaraX) [inline apps/web/src/api/api.client.ts]: is this `jwt` param needed here? I don't see it being used in the function

**Review by ChmaraX — APPROVED** [review]

File: apps/web/src/api/api.client.ts (hunk 1 above)
Comment (SokratisVidros) [reply apps/web/src/api/api.client.ts]: Good catch. I removed it.

File: apps/web/src/components/providers/AuthProvider.tsx [hunk 2]
```diff
@@ -60,14 +58,10 @@ export const useAuth = () => {
...
-  const token = IS_EE_AUTH_ENABLED ? eeAuthTokenCookie.get() : getCommunityAuthToken();
-
-  return token || '';
-}
-
-export function getTokenClaims(): IJwtClaims | null {
-  const token = getToken();
+export async function getToken() {
+  if (IS_EE_AUTH_ENABLED) {
+    return (await window?.Clerk?.session?.getToken()) || '';
```
Comment (LetItRock) [inline apps/web/src/components/providers/AuthProvider.tsx:63]: Somehow, for me, in my PR `window. Clerk` was a little buggy with probably race-conditons. But when used without the `window` object, it was working fine.

Comment (SokratisVidros) [reply apps/web/src/components/providers/AuthProvider.tsx:63]: Weird, I guess it has to do with React renderings.

**Pull request merged**